### PR TITLE
Miscellaneous improvements to DNS introspection policies in connectivity tests

### PR DESCRIPTION
**Pull request by giorio94**
Please review commit by commit, and refer to the individual messages for additional details.

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -296,7 +296,7 @@ Makefile* @cilium/build
 /cilium-cli/connectivity/builder/all_ingress_deny_from_outside.go @cilium/sig-encryption
 /cilium-cli/connectivity/builder/bgp_control_plane.go @cilium/sig-bgp
 /cilium-cli/connectivity/builder/cluster_entity_multi_cluster.go @cilium/sig-clustermesh
-/cilium-cli/connectivity/builder/dns_only.go @cilium/sig-clustermesh
+/cilium-cli/connectivity/builder/dns_only.go @cilium/fqdn
```
Comment (giorio94) [inline CODEOWNERS:299]: /cc @cilium/fqdn

**Review by nebril — APPROVED** [review]

**Review by pippolo84 — APPROVED** [review]
LGTM, thanks! :100:

**Review by marseel — APPROVED** [review]

File: cilium-cli/connectivity/builder/builder.go [hunk 2]
```diff
@@ -300,6 +300,7 @@ func renderTemplates(param check.Parameters) (map[string]string, error) {
 		"clientEgressL7HTTPMatchheaderSecretPortRangeYAML": clientEgressL7HTTPMatchheaderSecretPortRangeYAML,
 		"clientEgressL7HTTPExternalYAML":                   clientEgressL7HTTPExternalYAML,
 		"clientEgressNodeLocalDNSYAML":                     clientEgressNodeLocalDNSYAML,
+		"clientEgressOnlyDNSPolicyYAML":                    clientEgressOnlyDNSPolicyYAML,
```
Comment (gandro) [inline cilium-cli/connectivity/builder/builder.go:303]: This is nice! Unrelated to the PR (more a proposal for a follow-up), we should also feature gate the NodeLocalDNS and OpenShift exceptions (if we can detect them properly)

**Review by gandro — APPROVED** [review]
ACK for FQDN

File: cilium-cli/connectivity/builder/builder.go (hunk 2 above)
Comment (giorio94) [reply cilium-cli/connectivity/builder/builder.go:303]: Yep, makes sense, give that they are not needed in most cases (I'm not sure about how feature detection works for them, but I guess we already have something for NodeLocalDNS given that it has dedicated tests). I'm going to mark this comment as resolved for the moment, to allow merging this PR.

**Review by tklauser — APPROVED** [review]

**Pull request merged**